selftests/functional/serial/test_requirements.py

```diff
@@ -88,52 +103,87 @@ class BasicTest(TestCaseTmpDir):
+        return f"{AVOCADO} run {spawner} {path}"
+
+    def single_success(self, podman=False):
         with script.Script(os.path.join(self.tmpdir.name,
                                         'test_single_success.py'),
                            SINGLE_SUCCESS_CHECK) as test:
-            command = self.command % (AVOCADO, test.path)
+            command = self.get_command(test.path, podman)
             result = process.run(command, ignore_status=True)
             self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
             self.assertIn('PASS 1', result.stdout_text,)
             self.assertNotIn('bash', result.stdout_text,)
 
-    @unittest.skipUnless(os.getenv('CI'), skip_package_manager_message)
-    def test_single_fail(self):
+    def single_fail(self, podman=False):
         with script.Script(os.path.join(self.tmpdir.name,
                                         'test_single_fail.py'),
                            SINGLE_FAIL_CHECK) as test:
-            command = self.command % (AVOCADO, test.path)
+            command = self.get_command(test.path, podman)
             result = process.run(command, ignore_status=True)
             self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
             self.assertIn('PASS 0', result.stdout_text,)
             self.assertIn('SKIP 1', result.stdout_text,)
             self.assertNotIn('-foo-bar-', result.stdout_text,)
 
-    @unittest.skipUnless(os.getenv('CI'), skip_install_message)
-    def test_multiple_success(self):
+    def multiple_success(self, podman=False):
         with script.Script(os.path.join(self.tmpdir.name,
                                         'test_multiple_success.py'),
                            MULTIPLE_SUCCESS) as test:
-            command = self.command % (AVOCADO, test.path)
+            command = self.get_command(test.path, podman)
             result = process.run(command, ignore_status=True)
             self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
             self.assertIn('PASS 3', result.stdout_text,)
             self.assertNotIn('vim-common', result.stdout_text,)
 
-    @unittest.skipUnless(os.getenv('CI'), skip_install_message)
-    def test_multiple_fails(self):
+    def multiple_fails(self, podman=False):
         with script.Script(os.path.join(self.tmpdir.name,
                                         'test_multiple_fail.py'),
                            MULTIPLE_FAIL) as test:
-            command = self.command % (AVOCADO, test.path)
+            command = self.get_command(test.path, podman)
             result = process.run(command, ignore_status=True)
             self.assertEqual(result.exit_status, exit_codes.AVOCADO_ALL_OK)
             self.assertIn('PASS 1', result.stdout_text,)
             self.assertIn('SKIP 2', result.stdout_text,)
             self.assertNotIn('-foo-bar-', result.stdout_text,)
 
+    @unittest.skipUnless(os.getenv('CI'), skip_package_manager_message)
```

Comment:
These could later be converted into Avocado tests and use variants (instead of the repeated test with the `podman=True` parameters).

Another point to be discussed and considered, is if we could/should rely on requirements mechanism itself to determine if podman tests should run (if `podman` package is available?), instead of using `skip*`.

Reply:
This is a great idea, I will do that. I was so focused on the problem itself that I didn't use avocado features for making my testing easier :sweat_smile: